pkg/config/defaults.go

```diff
@@ -23,14 +23,16 @@ import (
 	"github.com/iancoleman/strcase"
 )
 
-// Common ExternalName configurations.
+// Commonly used resource configurations.
 var (
 	// NameAsIdentifier uses "name" field in the arguments as the identifier of
 	// the resource.
 	NameAsIdentifier = ExternalName{
 		SetIdentifierArgumentFn: func(base map[string]interface{}, name string) {
```

Comment:
nit: could we make it look similar to below configs by defining it as a function and using its name here.

Reply:
It's not really much meaningful for this function to live without the `OmittedFields` here as opposed the ones we have like `NopSetIdentifierArgument`, which is useful on its own.